File: backend/model/feature_guide.go
package model

type FeatureGuideSurface string

const (
	FeatureGuideSurfaceCanvas FeatureGuideSurface = "canvas"
	FeatureGuideSurfaceImage  FeatureGuideSurface = "image"
	FeatureGuideSurfaceVideo  FeatureGuideSurface = "video"
)

func (surface FeatureGuideSurface) Valid() bool {
	return surface == FeatureGuideSurfaceCanvas || surface == FeatureGuideSurfaceImage || surface == FeatureGuideSurfaceVideo
}

type FeatureGuide struct {
	BaseModel
	Surface FeatureGuideSurface `gorm:"size:20;uniqueIndex;not null" json:"surface"`
	Enabled bool                `gorm:"default:false;not null" json:"enabled"`
	Title   string              `gorm:"size:100;not null" json:"title"`
	Pages   string              `gorm:"type:longtext;not null" json:"-"`
	Version int                 `gorm:"default:1;not null" json:"version"`
}

func (FeatureGuide) TableName() string { return "feature_guides" }

type FeatureGuidePayload struct {
	Surface FeatureGuideSurface `json:"surface"`
	Enabled bool                `json:"enabled"`
	Title   string              `json:"title"`
	Pages   []string            `json:"pages"`
	Version int                 `json:"version"`
}

File: web/src/services/api/feature-guide.ts
import apiClient from "./client";

export type FeatureGuideSurface = "canvas" | "image" | "video";

export type FeatureGuide = {
    surface: FeatureGuideSurface;
    enabled: boolean;
    title: string;
    pages: string[];
    version: number;
};

export type SaveFeatureGuideInput = Pick<FeatureGuide, "enabled" | "title" | "pages">;

export async function getFeatureGuide(surface: FeatureGuideSurface): Promise<FeatureGuide | null> {
    const res = await apiClient.get(`/feature-guides/${surface}`);
    return res.data.data as FeatureGuide | null;
}

export async function getAdminFeatureGuides(): Promise<FeatureGuide[]> {
    const res = await apiClient.get("/admin/feature-guides");
    return res.data.data as FeatureGuide[];
}

export async function saveAdminFeatureGuide(surface: FeatureGuideSurface, input: SaveFeatureGuideInput): Promise<FeatureGuide> {
    const res = await apiClient.put(`/admin/feature-guides/${surface}`, input);
    return res.data.data as FeatureGuide;
}
